frontend/admin/resources/js/apiRoutes.ts

```diff
@@ -0,0 +1,35 @@
+import path from "path-browserify";
+
+export type ApiRoutes = ReturnType<typeof apiRoutes>;
+
+const apiRoutes = () => {
+  const apiRoot = (): string => "/";
+  return {
+    login: (from?: string, locale?: string): string => {
+      const searchTerms: string[] = [];
+      if (from) searchTerms.push(`from=${encodeURI(from)}`);
+      if (locale) searchTerms.push(`locale=${encodeURI(locale)}`);
+      const searchString = searchTerms.reduce((previousValue, currentValue) => {
+        if (previousValue) return `${previousValue}&${currentValue}`;
+        return currentValue;
+      }, "");
```

Comment:
I think you might have overthought this 😉  - isn't this just [array.join()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) ?
```suggestion
      const searchString = searchTerms.join("&");
```

Reply:
Haha, you're right!  I was originally trying to reduce a collection of key-value pairs but gave up.  I didn't take the next logical simplification step.